### PR TITLE
Split upgrade method

### PR DESCRIPTION
**Pull request by mjudeikis**
### Which issue this PR addresses:

Fixes VSTS 7829998

### What this PR does / why we need it:

Splits management patch method and clusterUpgrade to be separate methods. This way we can do management operations and avoid upgrading clusters.
Sets channel to stable.

### Test plan for issue:

Unti and mocking

### Is there any documentation that needs to be updated for this PR?

Yes, Will be updated once Geneva action code is shipped

### CODE REVIEW

File: pkg/frontend/adminactions/upgrade.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+			Image:   stream.PullSpec,
+		}
+
+		_, err = configClient.ConfigV1().ClusterVersions().Update(cv)
+		return err
+	})
+}
+
+func preUpgradeChecks(ctx context.Context, oc *api.OpenShiftCluster, vnet network.VirtualNetworksClient) error {
+	return checkCustomDNS(ctx, oc, vnet)
```
Comment (nilsanderselde) [inline pkg/frontend/adminactions/upgrade.go:70]: Are you anticipating needing to add more to this function? If not, I would just call `checkCustomDNS` directly from `Upgrade`.

File: pkg/frontend/adminactions/upgrade.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+	v, err := vnet.Get(ctx, r.ResourceGroup, r.ResourceName, "")
+	if err != nil {
+		return err
+	}
+
+	if v.VirtualNetworkPropertiesFormat.DhcpOptions != nil &&
+		v.VirtualNetworkPropertiesFormat.DhcpOptions.DNSServers != nil &&
+		len(*v.VirtualNetworkPropertiesFormat.DhcpOptions.DNSServers) > 0 {
+		return fmt.Errorf("not upgrading: custom DNS is set")
```
Comment (nilsanderselde) [inline pkg/frontend/adminactions/upgrade.go:99]: nice

**Comment by nilsanderselde** [comment]
I like it

File: pkg/cluster/disableupdates.go [hunk 3]
```diff
@@ -1,26 +0,0 @@
...
-// Licensed under the Apache License 2.0.
-
-import (
-	"context"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/util/retry"
-)
-
-func (i *manager) disableUpdates(ctx context.Context) error {
```
Comment (jim-minter) [inline pkg/cluster/disableupdates.go:13]: I'd love to get rid of this but I don't think we can.  Also note: there's no point adding the customDNS check if we're just going to have all the clusters on the stable channel...

File: pkg/util/version/const.go [hunk 4]
```diff
@@ -8,6 +8,7 @@ var GitCommit = "unknown"
 // InstallStream describes stream we are defaulting to for all new clusters
 var InstallStream = Stream{
 	Version:    NewVersion(4, 4, 17),
+	Channel:    "stable-4.4",
```
Comment (jim-minter) [inline pkg/util/version/const.go:11]: think these changes can be reverted for now ...

**Review by jim-minter — CHANGES_REQUESTED** [review]

File: pkg/frontend/adminactions/upgrade.go (hunk 1 above)
Comment (mjudeikis) [reply pkg/frontend/adminactions/upgrade.go:70]: I think there might be more of these... Hence new function.

File: pkg/cluster/disableupdates.go (hunk 3 above)
Comment (mjudeikis) [reply pkg/cluster/disableupdates.go:13]: so are we intentionally leaving cluster behind with this? Based on the new policy?

File: pkg/util/version/const.go (hunk 4 above)
Comment (mjudeikis) [reply pkg/util/version/const.go:11]: same as above

File: pkg/cluster/disableupdates.go (hunk 3 above)
Comment (mjudeikis) [reply pkg/cluster/disableupdates.go]: I was thinking about this more (needed more coffee).
 I disagree with this, but I disagree with the main reason we doing all this so this is wider disagreement :D 

Should we follow-up with this on MSFT scrum? as this does not sound right to me :/

**Comment by asalkeld** [comment]
looks good to me. also not this https://github.com/openshift/managed-upgrade-operator (operator to do pre checks, scale up a cluster so no down time, and monitor the progress of the upgrade). this is an effort started by dedicated SREs. Grant and I chatted with them yesterday.

**Review by jim-minter — APPROVED** [review]

**Pull request merged**